Tests/SecureXPCTests/XPCEncoder/Dictionary encoding tests.swift

```diff
@@ -0,0 +1,248 @@
+		
+		try assert(  dictOfUInt, encodesEqualTo: createXPCDict(fromOptionalElements:   dictOfUInt, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert( dictOfUInt8, encodesEqualTo: createXPCDict(fromOptionalElements:  dictOfUInt8, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert(dictOfUInt16, encodesEqualTo: createXPCDict(fromOptionalElements: dictOfUInt16, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert(dictOfUInt32, encodesEqualTo: createXPCDict(fromOptionalElements: dictOfUInt32, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert(dictOfUInt64, encodesEqualTo: createXPCDict(fromOptionalElements: dictOfUInt64, using: { xpc_uint64_create(UInt64($0)) }))
+
+		// Floating point numbers
+		let floats: [String: Float?] = ["float": 123, "nil": nil]
+		try assert(floats, encodesEqualTo: createXPCDict(fromOptionalElements: floats, using: { xpc_double_create(Double($0)) }))
+		let doubles: [String: Double?] = ["double": 123, "nil": nil]
+		try assert(doubles, encodesEqualTo: createXPCDict(fromOptionalElements: doubles, using: { xpc_double_create($0) }))
+
+		// Misc. types
+		
+		let bools: [String: Bool?] = ["false": false, "true": true, "nil": nil]
+		try assert(bools, encodesEqualTo: createXPCDict(fromOptionalElements: bools, using: { xpc_bool_create($0) }))
+		
+		// 💾
+		throw XCTSkip("Support for Data is not implemented, the contents just end up as a String.")
+		
+		let data: [String: Data?] = ["empty": Data(), "data": "Hello, world!".data(using: .utf8)!, "nil": nil]
+		try assert(data, encodesEqualTo: createXPCDict(fromOptionalElements: data, using: { datum in
+			datum.withUnsafeBytes { bufferP in xpc_data_create(bufferP.baseAddress, bufferP.count) }
+		}))
+		
+		// 📆
+		throw XCTSkip("Support for Date is not implemented, the contents just end up as a String.")
+		
+		let dates: [String: Date?] = ["distantPast": .distantPast, "now": .init(), "distantFuture": .distantFuture, "nil": nil]
+		try assert(dates, encodesEqualTo: createXPCDict(fromOptionalElements: dates, using: { date in
+			xpc_date_create(Int64(date.timeIntervalSince1970))
+		}))
+
+		let strings: [String: String?] = ["empty": "", "string": "Hello, world!", "nil": nil]
+		try assert(strings, encodesEqualTo: createXPCDict(fromOptionalElements: strings, using: { str in
+			// Does xpc_string_create copy the string? This would be a dangling pointer otherwise.
+			str.withCString(xpc_string_create)
+		}))
+		
+		throw XCTSkip("Support for UUID is not implemented, the contents just end up as a String.")
+		
+		let uuids: [String: UUID?] = ["uuid": UUID(), "nil": nil]
+		try assert(uuids, encodesEqualTo: createXPCDict(fromOptionalElements: uuids, using: { uuid in
+			var uuidByteTuple = uuid.uuid
+
+			return withUnsafePointer(to: &uuidByteTuple) { uuidByteTupleP in
+				uuidByteTupleP.withMemoryRebound(to: UInt8.self, capacity: 16) { uuidP in xpc_uuid_create(uuidP) }
+			}
+		}))
+
+		// TODO:
```

Comment:
Similar feedback as for arrays that as of now #3-8 don't have framework support, so a TODO about testing it probably doesn't belong here. FWIW I suspect that `XPCEncoder` wouldn't actually be involved in making any of these happen as I can't see how `Codable` could represent them.

Reply:
A simple property wrapper would do. I'll make an issue and we can discuss it there